gateway/testdata/valid.yaml

```diff
@@ -1,7 +1,11 @@
-basic:
-  username1:
-    ID: "orgid1"
-    password: "password1"
-  username2:
-    ID: "orgid2"
-    password: "password2"
\ No newline at end of file
+server-address: localhost:8080
```

Comment:
```suggestion
server:
  address: localhost
  port: 8080
```
what do you think?

it's very close to current cortex config:

```
server:
  http_listen_network: tcp
  http_listen_address: ""
  http_listen_port: 80
  http_listen_conn_limit: 0
  grpc_listen_network: tcp
  grpc_listen_address: ""
  grpc_listen_port: 9095
  grpc_listen_conn_limit: 0
  http_tls_config:
    cert_file: ""
    key_file: ""
    client_auth_type: ""
    client_ca_file: ""
  grpc_tls_config:
    cert_file: ""
    key_file: ""
    client_auth_type: ""
    client_ca_file: ""
  register_instrumentation: true
  graceful_shutdown_timeout: 30s
  http_server_read_timeout: 30s
  http_server_write_timeout: 30s
  http_server_idle_timeout: 2m0s
  grpc_server_max_recv_msg_size: 10485760
  grpc_server_max_send_msg_size: 10485760
  grpc_server_max_concurrent_streams: 10000
  grpc_server_max_connection_idle: 2562047h47m16.854775807s
  grpc_server_max_connection_age: 2562047h47m16.854775807s
  grpc_server_max_connection_age_grace: 2562047h47m16.854775807s
  grpc_server_keepalive_time: 2h0m0s
  grpc_server_keepalive_timeout: 20s
  grpc_server_min_time_between_pings: 10s
  grpc_server_ping_without_stream_allowed: true
  log_format: logfmt
  log_level: info
  log_source_ips_enabled: false
  log_source_ips_header: ""
  log_source_ips_regex: ""
  http_path_prefix: ""
```
https://cortexmetrics.io/docs/configuration/configuration-file/#server_config

Reply:
Yeah, that would be better. It would remove the need the parse the string as well